### PR TITLE
55 allow to upload remote rasters

### PR DESCRIPTION
**Pull request by ao-picterra**

### CODE REVIEW

File: src/picterra/__main__.py [hunk 1]
```diff
@@ -89,21 +92,75 @@ def parse_args(args):
...
+    create_raster_from_remote_parser = create_raster_subparsers.add_parser(
+        'remote', help="Uploads raster from a remote imagery server")
+    create_raster_from_remote_parser.add_argument(
+        "type", help="Type of the imagery server", type=str, choices=['wms', 'xyz'])
+    create_raster_from_remote_parser.add_argument("url", help="URL of the imagery server", type=str)
+    create_raster_from_remote_parser.add_argument(
+        "resolution", help="Spatial resolution (GSD) of the raster to upload, in meters",
+        type=float)
+
+    def create_remote_raster_type(string):
```
Comment (restaste) [inline src/picterra/__main__.py:126]: Perhaps rename `string` to something more explicit, like `coords_string`?

File: src/picterra/client.py [hunk 2]
```diff
@@ -204,6 +206,60 @@ def upload_raster(self, filename: str, name: str, folder_id=None,
...
+
+        Args:
+            server_type (str): One of ['wms', 'xyz']
+            server_url (str): URL for the imagery server
+            spatial_resolution_meters (float): GSD of the image we want to import
+            footprint (dict): GeoJSON Polygon containing the boundaries of the remote raster
+            credentials (optional, str): Credentials for the imagery server in the
+                "user:password" format, if needed
+            name (optional, str): Name the raster will take
+            folder_id (optional, str): Identifier of the folder this raster will belongs to
```
Comment (restaste) [inline src/picterra/client.py:225]: Perhaps mention here that if that's not provided, we'll put it in the `Picterra API Project` folder.

File: tests/test_client.py [hunk 3]
```diff
@@ -330,6 +344,23 @@ def test_upload_raster(identity_key):
...
+    add_mock_operations_responses('processing')
+    add_mock_operations_responses('success')
+    footprint = {
+        "type": "Polygon",
+        "coordinates": [ [ [10.0, 0.0], [11.0, 0.0], [11.0, 1.0],  [10.0, 1.0], [10.0, 0.0] ] ]
+    }
+    with pytest.raises(ValueError):
+        client.upload_remote_raster('spam', 'http://example.com', 0.2, footprint)
+    client.upload_remote_raster(type, 'http://example.com', 0.2, footprint)
+    assert len(responses.calls) == 3
```
Comment (restaste) [inline tests/test_client.py:361]: Perhaps just document here what the 3 expected calls are.

**Review by restaste — APPROVED** [review]
LGTM! A couple of doc nitpicks.

File: src/picterra/client.py (hunk 2 above)
Comment (ao-picterra) [reply src/picterra/client.py:225]: Checked (also updated the docstring of `upload_raster` too)

File: src/picterra/__main__.py (hunk 1 above)
Comment (ao-picterra) [reply src/picterra/__main__.py]: Checked

File: tests/test_client.py (hunk 3 above)
Comment (ao-picterra) [reply tests/test_client.py:362]: Checked

**Comment by julienr** [comment]
Should we merge this ?

**Comment by ao-picterra** [comment]
@restaste can we merge this?

**Comment by restaste** [comment]
LGTM! There are just 2 conflicts to fix.

**Pull request merged**